### ISSUE TITLE
Multiple Matches are not Displayed Correctly in the Report

### ISSUE THREAD

**Issue by parsiya** [microsoft/ApplicationInspector#183]
**Describe the bug**
If we use the `-d` option to allow duplicates all matches are not displayed in the report.

See the following test file (also attached as md5.py.txt):

```python
# 1: testing MD5
# 2:
3: MD5
# 4: some filler
5: MD5
# 6: some filler
7: MD5
# 8: more filler
9: MD5
# 10: end testing MD5
```

The application should detect 6 matches for MD5 (note it also detects MD5 in the comments which is fine). After running the report, the report only displays one match with invalid line numbers.

![image](https://user-images.githubusercontent.com/3981231/77816994-b4d3e100-7084-11ea-890d-b6d1cfe2ef95.png)

There are three issues:

1. Not all matches are displayed.
2. Line numbers start from -1.
3. The `Line Number` at the bottom is for the last match (line 10).

Looking inside the json file (also attached as `output.json.txt` because github does not support attaching json or html files). Remember to remove the `.txt` extension from the py files otherwise they will be skipped.

* [md5.py.txt](https://github.com/microsoft/ApplicationInspector/files/4396743/md5.py.txt)
* [output.json.txt](https://github.com/microsoft/ApplicationInspector/files/4396740/output.json.txt)
* [output.html.txt](https://github.com/microsoft/ApplicationInspector/files/4396741/output.html.txt)

Interestingly, if I remove the `MD5` in the first line. The line numbers are displayed correctly but the other two issues remain.

```python
# 1: testing
# 2:
3: MD5
# 4: some filler
5: MD5
# 6: some filler
7: MD5
# 8: more filler
9: MD5
# 10: end testing
```
![image](https://user-images.githubusercontent.com/3981231/77817230-9cfd5c80-7086-11ea-96d4-8802fb144728.png)

Originally, I used it on some code with multiple matches. In that case:

1. All matches were displayed.
2.  The `Line Number` at the bottom is for the last match regardless of which match is displayed. I expected it to change when I click on each match.
3. The first row was highlighted regardless of what the detected line number is for all matches (including the last)

I tried to replicate it and I got to this example.

PS: I understand that the tool was optimized to be used to show one match. However, multiple matches are supported. My usecase is not exactly the same as what is intended for the application. I use it for security to show me where issue X might be present and multiple matches are needed.

**Expected behavior**

1. Show all matches.
2. Display the correct line numbers (e.g., not `-1`).
3. When selecting a match, update the line number at the bottom to reflect the selected match.

**Comment by guyacosta** [comment]
The tool should show multiple matches on request.  A bug may have been introduced.  Investigating.  Thanks for submitting.

**Comment by guyacosta** [comment]
Partial fix.  The duplicates are actually detected the problem was UI related.  

1. The line numbers being negative for matches on the first 1-2 lines is fixed and still works normally for other lines.  This was unrelated to duplicates and simply an error for matches at the top.
2. The problem with supporting multiple matches in the HTML output is not addressed at this time other than to detect and block use of the duplicates feature for html format selection.  To get all non-unique matches in your output, use a different output format option like text or json.
